docs/troubleshooting.md

```diff
@@ -63,20 +79,189 @@ oc exec <es_pod> -c elasticsearch -- \
+
+The key in deleting documents from an index is in constructing appropriate ES query that matches all required documents. To learn how to write specific queries consult ES Query DSL documentation specifically the part about [query filter context](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/query-filter-context.html).
+
+We recommend testing the query first (Kibana [Dev Tools Console](https://www.elastic.co/guide/en/kibana/6.8/console-kibana.html) is a handy tool) before it is used in the `_delete_by_query` API context. 
+
+For example to match all **infra** logs from both the "**openshift-cluster-version**" and "**openshift-kube-apiserver**" namespaces you can use the following query:
+```json
+GET /infra/_search
+{
+  "query" : {
+    "terms": {
+      "kubernetes.namespace_name": [
+        "openshift-cluster-version",
+        "openshift-kube-apiserver"
+      ]
+    }
+  }
+}
 ```
-oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=.security/_settings?pretty -X PUT -d '{"index.blocks.read_only_allow_delete": null}'
+
+To match only those logs that are **older than 24 rhs** you need to add additional [range filter](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/query-dsl-range-query.html) into this query:
+```json
+GET /infra/_search
+{
+  "query": {
+    "bool": {
+      "must": [
+        {
+          "terms": {
+            "kubernetes.namespace_name": [
+              "openshift-cluster-version",
+              "openshift-kube-apiserver"
+            ]
+          }
+        }
+      ],
+      "filter": [
+        {
+          "range": {
+            "@timestamp": { "lt": "now-24h" }    // <- older than 24 hrs, lt = less-than
+          }
+        }
+      ]
+    }
+  }
+}
 ```
+Once the query is prepared, it is used to delete the documents. All it takes it just changing the very first line of the snippet from `GET <...>/_search` to `POST <...>/_delete_by_query`:
 
-You can also manually unblock all indices by using this command:
+```json
+POST /infra/_delete_by_query
+{
+  "query": {    // <- this is the same query
+    ... 
+  }
+}
+
+# If you face this error in Kibana UI then you shall execute the query from CLI
+{
+  "type": "security_exception",
+  "reason": "no permissions for [indices:data/write/delete/byquery] and User [name=kube:admin, roles=[admin_reader], requestedTenant=__user__]"
+}
 ```
-oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_all/_settings?pretty -X PUT -d '{"index.blocks.read_only_allow_delete": null}'
-```
\ No newline at end of file
+
+CLI approach would be (again, the `query.json` file must include only the JSON data, not the first line starting with `POST ...`):
+```shell
+QUERY=`cat query.json`; \
```

Comment:
can you have an example of the `query.json` file based on the above example for completeness?

Reply:
I added and linked json file examples in both CLI sue cases.